src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php

```diff
@@ -0,0 +1,709 @@
+     * @param TranslatorInterface $translator
+     * @param Context $context
+     */
+    public function __construct(
+        ImageTagSourceParserInterface $imageTagSourceParser,
+        TranslatorInterface $translator,
+        Context $context
+    ) {
+        $this->imageTagSourceParser = $imageTagSourceParser;
+        $this->translator = $translator;
+        $this->context = $context;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(GetOrderForViewing $query): OrderForViewing
+    {
+        $order = $this->getOrder($query->getOrderId());
+
+        $taxMethod = $order->getTaxCalculationMethod() == PS_TAX_EXC ?
+            $this->translator->trans('Tax excluded', [], 'Admin.Global') :
+            $this->translator->trans('Tax included', [], 'Admin.Global');
+
+        return new OrderForViewing(
+            (int) $order->id,
+            (int) $order->id_currency,
+            $order->reference,
+            $taxMethod,
+            (bool) $order->valid,
+            $this->getOrderCustomer($order),
+            $this->getOrderShippingAddress($order),
+            $this->getOrderInvoiceAddress($order),
+            $this->getOrderProducts($order),
+            $this->getOrderHistory($order),
+            $this->getOrderDocuments($order),
+            $this->getOrderShipping($order),
+            $this->getOrderReturns($order),
+            $this->getOrderPayments($order),
+            $this->getOrderMessages($order)
+        );
+    }
+
+    /**
+     * @param OrderId $orderId
+     *
+     * @return Order
+     *
+     * @throws OrderNotFoundException
+     */
+    private function getOrder(OrderId $orderId): Order
+    {
+        $order = new Order($orderId->getValue());
+
+        if ($order->id !== $orderId->getValue()) {
+            throw new OrderNotFoundException(
+                $orderId,
+                sprintf('Order with id "%s" was not found.', $orderId->getValue())
+            );
+        }
+
+        return $order;
+    }
+
+    /**
+     * @param Order $order
+     *
+     * @return OrderCustomerForViewing
+     */
+    private function getOrderCustomer(Order $order): OrderCustomerForViewing
+    {
+        $customer = new Customer($order->id_customer);
+        $gender = new Gender($customer->id_gender);
+        $genderName = '';
+
+        if (Validate::isLoadedObject($gender)) {
+            $genderName = $gender->name[$order->id_lang];
+        }
+
+        $customerStats = $customer->getStats();
+
+        return new OrderCustomerForViewing(
+            $customer->id,
+            $customer->firstname,
+            $customer->lastname,
+            $genderName,
+            $customer->email,
+            new DateTimeImmutable($customer->date_add),
+            Tools::displayPrice(Tools::ps_round(Tools::convertPrice($customerStats['total_orders'], $order->id_currency), PS_ROUND_HALF_UP), (int) $order->id_currency),
```

Comment:
`Tools::displayPrice()` is deprecated, use `Locale::formatPrice()` instead

Reply:
We can do it in a following PR

Reply:
I'll mark it as TODO in issue.